.github/workflows/build.yml

```diff
@@ -0,0 +1,119 @@
+name: Publish and Deploy
+run-name: Publish and Deploy (PDFs) on PR Merge
+
+# Definizione dei permessi
+permissions:
+  contents: write
+  pages: write
+  id-token: write
+
+# Trigger: pull request sul ramo base
+on:
+  pull_request:
+    branches: 
+      - main
+    types: [closed]
+
+# Unità di lavoro all'interno del workflow
+jobs:
+  docker:
+  
+    # Controllo se la pull request è stata unita al ramo base
+    if: github.event.pull_request.merged == true
+    
+    # Esecuzione in ambiente Ubuntu con le ultime versioni di pacchetti e librerie
+    runs-on: ubuntu-latest
+    
+    # Azioni atomiche eseguite in sequenza all'interno del job
+    steps:
+    
+    # Clonazione del repository sorgente all'interno dell'ambiente di esecuzione del job
+    - name: Checkout Source Repository
+      uses: actions/checkout@v4
+      with:
+          # Recupero tutta la cronologia (branch e tag)
+          fetch-depth: 0
+
+    # Salvataggio e ripristino della cache
```

Comment:
Sul caching di Docker:
Al momento questa soluzione specifica come contenuto della cache solo il contenuto di `tools/docker`, che sono solamente file interni alla repo. Allo scopo di evitare di dover ricostruire l'immagine ogni volta si potrebbe migliorare questa componente di action ricorrendo a registri di container, come [Docker Hub](hub.docker.com) o [Gitub Container Registry](ghcr.io). Lo giudico uno sforzo aggiuntivo che può far comodo, ma ci si può pensare in futuro senza bloccare questa PR